R/labels.R

```diff
@@ -84,6 +84,15 @@ setup_plot_labels <- function(plot, layers, data) {
     ))
   }
 
+  dict <- plot_labels$dictionary
+  if (length(dict) > 0) {
+    labels <- lapply(labels, function(x) {
+      dict <- dict[names(dict) %in% x]
+      x[match(names(dict), x)] <- dict
+      x
+    })
+  }
+
```

Comment:
This is being called on all labels, right? So if a user has given a label name that just happens to be in the dictionary it will get translated

To me it should only apply to fallback labels, but I'm open to hearing against this

Reply:
Can't roll back my approval but let's talk this one out before merging :-)

Reply:
It applies to all 'derived' labels, not labels that users give verbatim.
Is this the kind of situation you mean?

``` r
devtools::load_all("~/packages/ggplot2/")
#> ℹ Loading ggplot2

ggplot(mpg, aes(displ, hwy)) +
  geom_point() +
  labs(
    x = "Engine Displacement",
    dictionary = c(displ = "NO NOT THIS ONE", hwy = "new label")
  )
```

![](https://i.imgur.com/pWsdPMN.png)<!-- -->

<sup>Created on 2024-12-02 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
Also `labs(x = "foo", dictionary = c(foo = "bar"))` will give `"foo"` as label.